packages/accounts/README.md

```diff
@@ -267,3 +267,47 @@ const myDecoder: Decoder<MyAccountData> = getStructDecoder([
 const myDecodedAccount = decodeAccount(myAccount, myDecoder);
 myDecodedAccount satisfies Account<MyAccountData, '1234..5678'>;
 ```
+
+### `assertAccountDecoded()`
+
+This function asserts that an account stores decoded data, ie not a Uint8Array. Note that it does not check the shape of the data matches the decoded type, only that it is not a Uint8Array.
+
+```ts
+type MyAccountData = { name: string; age: number };
+
+const myAccount: Account<MyAccountData | Uint8Array, '1234..5678'>;
+assertAccountDecoded(myAccount);
+
+// now the account data can be used as MyAccountData
+account.data satisfies MyAccountData;
+```
+
+This is particularly useful for narrowing the result of fetching a JSON parsed account.
+
+
+```ts
+    const account: MaybeAccount<MockData | Uint8Array> = await fetchJsonParsedAccount<MockData>(
+        rpc, 
+        '1234..5678' as Address,
+    )
+
+    assertAccountDecoded(account);
+    // now we have a MaybeAccount<MockData>
+    account satisfies MaybeAccount<MockData>
+```
```

Comment:
Wouldn't this make sense to return a `Account<MockData>`? If you can verify the data is not a `Uint8Array`, then you are also confirming its existence, no?

Reply:
I think it makes more sense to let the two different type guards check their respective properties so they can be composed.

> If you can verify the data is not a Uint8Array, then you are also confirming its existence, no?

No, if the account does not exist it does not fail when asserting it's decoded.

Reply:
Yep, the main reason it works like this (if no data then assert is OK) is so you can compose the checks in either order.

It also mirrors the behaviour of `decodeAccount`, which will no-op on an account that doesn't exist, ie will convert `MaybeAccount<Uint8Array>` to `MaybeAccount<TData>`. I think it'd be really confusing if this threw after you call `decodeAccount`:

```ts
const account = <MaybeEncodedAccount>{address: '1111' as Address, exists: false};

// this already works
const decoder: Decoder<TData> = ...
const decodedAccount = decodeAccount(account, decoder)

// this should also
assertAccountDecoded(decodedAccount)
```

Reply:
Alrighty sounds good!